source/multigrid/mg_tools.cc

```diff
@@ -1547,6 +1547,32 @@ namespace MGTools
       }
 
   }
+
+
+
+  template <int dim, int spacedim>
+  unsigned int
+  max_level_for_coarse_mesh (const Triangulation<dim,spacedim> &tria)
+  {
+    // Find minimum level for an active cell in
+    // this locally owned subdomain
+    int min_level = tria.n_global_levels();
```

Comment:
Unsigned int

Reply:
I originally had `unsigned int`, however you receive warnings when you have 
`cell->level() < min_level` below since the function `level()` can be negative and therefore returns an `int`

Reply:
The fact that `level` returns a signed int is a historical mistake. Just cast the returned value to `unsigned int` and make the variable unsigned as well.